### PR TITLE
test: use MiniWallet for make_utxo helper in feature_rbf.py

### PR DESCRIPTION
**Pull request by theStack**
This PR aims to further increase MiniWallet usage in the functional test feature_rbf.py by using it in the `make_utxo(...)` helper, which is the only part that needs a wallet for most sub-tests. In order to do that, the amounts for the utxos have to be scaled down in two sub-tests first (`test_doublespend_chain` and `test_doublespend_tree`, see first two commits), since we need amounts passed to `make_utxo` than can be funded by only one input. For creating UTXOs with a value of 50 BTC, we'd need to implement a method for consolidating multiple utxos into one first, which seems to be overkill.

Note that after this PR's change, there is only one sub-test left (`test_rpc`) that needs the wallet compiled into bitcoind.

### CODE REVIEW

**Comment by Shubhankar-Gambhir** [comment]
concept ACK
successfully ran test with 
<details>
<summary> commenting test_rpc subtest </summary>

```
2021-09-21T10:20:52.294000Z TestFramework (INFO): Initializing test directory /tmp/bitcoin_func_test_gvoms1s4
2021-09-21T10:20:57.334000Z TestFramework (INFO): Running test simple doublespend...
2021-09-21T10:20:57.352000Z TestFramework (INFO): Running test doublespend chain...
2021-09-21T10:20:57.415000Z TestFramework (INFO): Running test doublespend tree...
2021-09-21T10:20:57.877000Z TestFramework (INFO): Running test replacement feeperkb...
2021-09-21T10:20:57.934000Z TestFramework (INFO): Running test spends of conflicting outputs...
2021-09-21T10:20:57.945000Z TestFramework (INFO): Running test new unconfirmed inputs...
2021-09-21T10:20:57.953000Z TestFramework (INFO): Running test too many replacements...
2021-09-21T10:20:58.020000Z TestFramework (INFO): Running test opt-in...
2021-09-21T10:20:58.034000Z TestFramework (INFO): Running test prioritised transactions...
2021-09-21T10:20:58.114000Z TestFramework (INFO): Running test no inherited signaling...
2021-09-21T10:20:58.121000Z TestFramework (INFO): Check that the child tx can still be replaced (via a tx that also replaces the parent)
2021-09-21T10:20:58.124000Z TestFramework (INFO): Running test replacement relay fee...
2021-09-21T10:20:58.126000Z TestFramework (INFO): Passed
2021-09-21T10:20:58.177000Z TestFramework (INFO): Stopping nodes
2021-09-21T10:20:58.581000Z TestFramework (INFO): Cleaning up /tmp/bitcoin_func_test_gvoms1s4 on exit
2021-09-21T10:20:58.582000Z TestFramework (INFO): Tests successful
```
</details>

<details>
<summary> changes on  test_rpc proposed in #23052  </summary>

```
2021-09-21T10:18:35.765000Z TestFramework (INFO): Initializing test directory /tmp/bitcoin_func_test_q9cpp4nw
2021-09-21T10:18:42.487000Z TestFramework (INFO): Running test simple doublespend...
2021-09-21T10:18:42.504000Z TestFramework (INFO): Running test doublespend chain...
2021-09-21T10:18:42.553000Z TestFramework (INFO): Running test doublespend tree...
2021-09-21T10:18:43.002000Z TestFramework (INFO): Running test replacement feeperkb...
2021-09-21T10:18:43.062000Z TestFramework (INFO): Running test spends of conflicting outputs...
2021-09-21T10:18:43.073000Z TestFramework (INFO): Running test new unconfirmed inputs...
2021-09-21T10:18:43.081000Z TestFramework (INFO): Running test too many replacements...
2021-09-21T10:18:43.147000Z TestFramework (INFO): Running test opt-in...
2021-09-21T10:18:43.159000Z TestFramework (INFO): Running test RPC...
2021-09-21T10:18:43.728000Z TestFramework (INFO): Running test prioritised transactions...
2021-09-21T10:18:43.817000Z TestFramework (INFO): Running test no inherited signaling...
2021-09-21T10:18:43.823000Z TestFramework (INFO): Check that the child tx can still be replaced (via a tx that also replaces the parent)
2021-09-21T10:18:43.825000Z TestFramework (INFO): Running test replacement relay fee...
2021-09-21T10:18:43.828000Z TestFramework (INFO): Passed
2021-09-21T10:18:43.878000Z TestFramework (INFO): Stopping nodes
2021-09-21T10:18:47.246000Z TestFramework (INFO): Cleaning up /tmp/bitcoin_func_test_q9cpp4nw on exit
2021-09-21T10:18:47.246000Z TestFramework (INFO): Tests successful
```
</details>

File: test/functional/feature_rbf.py [hunk 1]
```diff
@@ -89,29 +96,23 @@ def run_test(self):
...
-        txid = node.sendrawtransaction(signed_tx['hex'], 0)
+        # MiniWallet only supports sweeping utxos to its own internal scriptPubKey, so in
+        # order to create an output with arbitrary amount/scriptPubKey, we have to add it
+        # manually after calling the create_self_transfer method. The MiniWallet output's
+        # nValue has to be adapted accordingly (amount and fee deduction). To keep things
+        # simple, we use a fixed fee of 1000 Satoshis here.
+        fee = 1000
+        tx = self.wallet.create_self_transfer(from_node=node, fee_rate=0, mempool_valid=False)['tx']
+        assert_greater_than(tx.vout[0].nValue, amount + fee)
+        tx.vout[0].nValue -= (amount + fee)           # change output -> MiniWallet
```
Comment (MarcoFalke) [inline test/functional/feature_rbf.py:113]: What happens if this output is removed? Previously the wallet would also "donate" the change to the fee.

**Review by maflcko — APPROVED** [review]
review ACK f680d27155374de658d40db0ba40460919aa1ba2 🦐

<details><summary>Show signature and timestamp</summary>

Signature:

```
-----BEGIN PGP SIGNED MESSAGE-----
Hash: SHA512

review ACK f680d27155374de658d40db0ba40460919aa1ba2 🦐
-----BEGIN PGP SIGNATURE-----

iQGzBAEBCgAdFiEE+rVPoUahrI9sLGYTzit1aX5ppUgFAlwqrYAACgkQzit1aX5p
pUjbcgwAkRTDpqwf/UETFI6ocdBFK2PZqYgMFszvH/UuC2h2MvgeZGV8ZnBl8BKQ
4wyWuZJSY13QCjVbaUeF0jqzMw8mJd1KUMaAw3nHP4lasGpuOxhhP5l0xyK6UJB1
1inOWwb1GEB2p5Bhd6m0q4ucSo3da5OBdE1OO+GAmOBaEZE5AKvExIxnZEd74A1E
yL1K6vzw+ajJaofIOm/AkOe9x7mrHC4ilqOrQVf4UHZ3YG9np9BVDx4srCbyaXWr
OHeTY9EJx38XSLJooQs9Ap+7zexx8jXv6vqsJuR+deZoq83xqo4WaInrXsCFDiux
yFccruXkXlIh47BEkwSKRBI/7avqiNvNoVKiQGHhFbn2km9YIkEJRASfCuB5kxca
MM6xt6GfvSQc7IiUiN3DaACJH1ut5GALVX0KiJRw1Y+wYlWJjqwoG0TgJUezXaUy
5+oP93maZNRsWses07raJXmJ8j2hSYcvPmfGeOKWDgDpEXJfk/r7b/MZFPBzOG8q
kqgKZP7P
=4DjW
-----END PGP SIGNATURE-----
```

[Timestamp of file with hash `8c9f3fb0e7a45d86ca12d77967ff62c2f410f3766b9ea0920fb291caba583a61  -`](https://opentimestamps.org/info/?004f70656e54696d657374616d7073000050726f6f6600bf89e2e884e8929401088c9f3fb0e7a45d86ca12d77967ff62c2f410f3766b9ea0920fb291caba583a61f010ed67ccb8e20b6de88cc6191122da1c8608fff01019f9d4d6fda9d5c02f0efff91232d7bf08f0202a12f36a1135689f9363c10a7cf0075c725b31afca04899352a24f66abfaba5208f0208e0df4261d7da7b6bcc393c10db916a5d6256b7ac3b87944beaa9a9a27aa1eff08f1046149b9c8f0088c608d5c6c31648a0083dfe30d2ef90c8e2e2d68747470733a2f2f616c6963652e6274632e63616c656e6461722e6f70656e74696d657374616d70732e6f7267fff01046211c62239bc4c767ff059ace2336c808f1046149b9c8f0089bc4f53de046089b0083dfe30d2ef90c8e2c2b68747470733a2f2f626f622e6274632e63616c656e6461722e6f70656e74696d657374616d70732e6f7267fff0108e36c1e167056798bef88409ebf6bfec08f020295fa3ff3e2aae7a90219419ba18d6d7877c72c22e332bb6f5e86e2c53720fcf08f020cbf8808fe1ae4400b2054b20fa407aad4995e018c387e485a1b4a3393121fe2c08f1046149b9c8f00866fad77dd4335bc30083dfe30d2ef90c8e292868747470733a2f2f66696e6e65792e63616c656e6461722e657465726e69747977616c6c2e636f6df010abecb901e0bd820c3964d035e6e85a8908f1046149b9c8f0080c5ae829b925269e0083dfe30d2ef90c8e232268747470733a2f2f6274632e63616c656e6461722e636174616c6c6178792e636f6d)

</details>

File: test/functional/feature_rbf.py (hunk 1 above)
Comment (theStack) [reply test/functional/feature_rbf.py:113]: IIUC what you mean: without this output, no new utxo to the internal MiniWallet scriptPubKey is created, i.e. we would probably run out of funds (or, could only create as many transactions as there are coinbase outputs from the pre-mined blocks from the test framework).

**Pull request merged**